### PR TITLE
Refactor slotpicker

### PR DESCRIPTION
**Pull request by chrishanes**
The calendar JS was far too long and in one file. Refactoring the code and breaking out into multiple files for the separate parts of the process for choosing a date and time.

### CODE REVIEW

File: app/assets/javascripts/modules/slotpicker/moj.datepicker.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        moj.Modules.Slotpicker.SlotSource.removeSlot();
+        var form = $(this).parents('form');
+        form.submit();
+      });
+
+      this.$skipBtn.on('click', function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var form = $(this).parents('form');
+        $('#slots_step_skip_remaining_slots').val(true);
```
Comment (StupidCodeFactory) [inline app/assets/javascripts/modules/slotpicker/moj.datepicker.js]: Could this be an `this.$myId`?

File: app/assets/javascripts/modules/slotpicker/_calendar.js [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+
+      gridCells += '\t</tr>';
+      this.addRows(gridCells);
+      this.setPrevBtn();
+      this.setNextBtn();
+    },
+
+    clearGrid: function() {
+      this.$tbody.empty();
+      $('#msg').empty();
```
[the remark this answers is not in the document]
Comment (chrishanes) [reply app/assets/javascripts/modules/slotpicker/_calendar.js]: This is only referenced once, so doesn't need to be cached

File: app/assets/javascripts/modules/slotpicker/moj.datepicker.js (hunk 1 above)
Comment (chrishanes) [reply app/assets/javascripts/modules/slotpicker/moj.datepicker.js]: Again, this is an element that is referenced once so doesn't need to be cached.

File: app/assets/javascripts/modules/slotpicker/_helpers.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    timeFromSlot: function(slot) {
+      var time = new Date();
+      time.setHours(slot.split(':')[0]);
+      time.setMinutes(slot.split(':')[1]);
+      return time;
+    },
+
+    getUnique: function(arr) {
+      var unique = [];
+      for (var i = 0; i < arr.length; i++) {
```
Comment (TobyRet) [inline app/assets/javascripts/modules/slotpicker/_helpers.js]: Can Array.prototype.filter() be used here?

**Pull request merged**